### PR TITLE
#3690 fix issue with skipped steps

### LINKED ISSUE #3690: [Bug]: Some variables not copied correctly if experiment steps are skipped
### PyBaMM Version

23,9

### Python Version

3.11

### Describe the bug

Sometimes experiement steps are skipped (e.g. skip a charge if you're already at that voltage). When this happens the solver returns an `EmptySolution`. The next step of the experiment then starts from the wrong point.

### Steps to Reproduce

```python3
import pybamm
import matplotlib.pyplot as plt

model = pybamm.lithium_ion.SPMe(
    {
        "SEI": "solvent-diffusion limited",
    }
)

exp = pybamm.Experiment(
    [(f"Rest for 24 hours (1 hour period)",)]
    + [
        (
            "Charge at C/3 until 4.1 V",
            "Hold at 4.1V until C/20",
            "Discharge at C/3 until 2.5 V",
        )
    ]
)

sim1 = pybamm.Simulation(model, experiment=exp)
sim1.solve(initial_soc=1)
sim2 = pybamm.Simulation(model, experiment=exp)
sim2.solve(initial_soc=0.9)

sols = [sim1.solution, sim2.solution]
labels = ["SOC=1", "SOC=0.9"]
fig, ax = plt.subplots(2, 2, figsize=(10, 4))
for sol, label in zip(sols, labels):
    ax[0,0].plot(sol["Time [h]"].data, sol["Terminal voltage [V]"].data, label=label)
    ax[0,1].plot(sol["Time [h]"].data, sol["X-averaged negative particle surface concentration [mol.m-3]"].data, label=label)
    ax[1,0].plot(sol["Time [h]"].data, sol["Total capacity lost to side reactions [A.h]"].data, label=label)
    ax[1,1].plot(sol["Time [h]"].data, sol["X-averaged negative total SEI thickness [m]"].data, label=label)
ax[0, 0].legend()

sim1.solution.cycles[1].steps
sim2.solution.cycles[1].steps
```

image

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by rtimms**
# Description
Fixed a bug where if the first step(s) in a cycle are skipped then the cycle solution started from the model's initial conditions instead of from the last state of the previous cycle.

Fixes #3690 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by rtimms** [comment]
@Saransh-cpp this is a bug fix that should go into 24.1

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # [Unreleased](https://github.com/pybamm-team/PyBaMM/)
 
+## Bug Fixes
+
+- Fixed a bug where if the first step(s) in a cycle are skipped then the cycle solution started from the model's initial conditions instead of from the last state of the previous cycle ([#3708](https://github.com/pybamm-team/PyBaMM/pull/3708))
```
Comment (tinosulzer) [inline CHANGELOG.md:5]: if this is going into 24.1 this needs to move

**Review by tinosulzer — APPROVED** [review]
Looks good thanks

File: CHANGELOG.md (hunk 1 above)
Comment (rtimms) [reply CHANGELOG.md:5]: what’s the process? should i just manually make a new section for the next release candidate?

Comment (Saransh-cpp) [reply CHANGELOG.md:5]: You can leave it as it is. The workflow will automatically create a v24.1rc1 section with the new changes.

Comment (rtimms) [reply CHANGELOG.md:5]: cool. where/when should this be merged?

**Pull request merged**